cmd/podman/images.go

```diff
@@ -92,68 +90,67 @@ func imagesCmd(c *cli.Context) error {
 		return errors.Wrapf(err, "Could not get runtime")
 	}
 	defer runtime.Shutdown(false)
-
-	format := genImagesFormat(c.String("format"), c.Bool("quiet"), c.Bool("noheading"), c.Bool("digests"))
-
-	opts := imagesOptions{
-		quiet:     c.Bool("quiet"),
-		noHeading: c.Bool("noheading"),
-		noTrunc:   c.Bool("no-trunc"),
-		digests:   c.Bool("digests"),
-		format:    format,
-	}
-
+	var filterFuncs []libpod.ImageResultFilter
 	var imageInput string
 	if len(c.Args()) == 1 {
 		imageInput = c.Args().Get(0)
+		//image := runtime.NewImage(imageInput)
+		//_, imageID, _ = image.GetLocalImageName()
+
+		//if image.LocalName == "" {
+		//	return errors.Errorf("unable to determine local image %s", imageInput)
+		//}
 	}
 	if len(c.Args()) > 1 {
 		return errors.New("'podman images' requires at most 1 argument")
 	}
 
-	params, err := runtime.ParseImageFilter(imageInput, c.String("filter"))
-	if err != nil {
-		return errors.Wrapf(err, "error parsing filter")
+	if len(c.StringSlice("filter")) > 0 || len(strings.TrimSpace(imageInput)) != 0 {
+		filterFuncs, err = libpod.CreateFilterFuncs(runtime, c, imageInput)
+		if err != nil {
+			return err
+		}
 	}
 
-	// generate the different filters
-	labelFilter := generateImagesFilter(params, "label")
-	beforeImageFilter := generateImagesFilter(params, "before-image")
-	sinceImageFilter := generateImagesFilter(params, "since-image")
-	danglingFilter := generateImagesFilter(params, "dangling")
-	referenceFilter := generateImagesFilter(params, "reference")
-	imageInputFilter := generateImagesFilter(params, "image-input")
-
-	/*
-		podman does not implement --all for images
```

Comment:
@baude Did you mean to get rid of this? I don't see a --all CLI flag added yet?

Reply:
nope, just got caught up in all changes